appserver/payara-appserver-modules/microprofile/opentracing/src/main/java/fish/payara/microprofile/opentracing/jaxrs/JaxrsContainerRequestTracingFilter.java

```diff
@@ -153,9 +166,21 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
                     && requestTracing.isRequestTracingEnabled()
                     && requestTracing.isTraceInProgress()) {
 
-                // Get the traced annotation from the method
-                Traced tracedAnnotation = OpenTracingCdiUtils.getAnnotation(CDI.current().getBeanManager(),
-                        Traced.class, resourceInfo);
+                // Check if CDI has been initialised by trying to get the BeanManager
+                BeanManager beanManager = null;
+                try {
+                    beanManager = CDI.current().getBeanManager();
+                } catch (IllegalStateException ise) {
+                    // *Should* only get here if CDI hasn't been initialised, indicating that the app isn't using it
+                    logger.log(Level.FINE, "Error getting Bean Manager, presumably due to this application not using CDI", 
```

Comment:
Is this log message not going to get logged twice?

Reply:
Yes, but it is possible for the two filter methods to execute on separate threads.